packages/discovery-react-components/src/utils/useDataApi.ts

```diff
@@ -196,12 +198,13 @@ export const useFieldsApi = (
     searchClient.listFields,
     searchClient
   );
+
+  const fetchFields = useCallback(() => setFetchToken({ trigger: true }), [setFetchToken]);
```

Comment:
These type of callbacks were getting recreated on every render (since they were anonymous functions created within an object definition). Instead, wrap them with `useCallback` to prevent re-creation.